### ISSUE TITLE
Calling `useRoute` within middleware may lead to misleading results

### ISSUE THREAD

**Issue by andrzejkupczyk** [harlan-zw/nuxt-link-checker#37]
### Describe the bug

I'm seeing this warning in dev console:

> [nuxt] Calling `useRoute` within middleware may lead to misleading results. Instead, use the (to, from) arguments passed to the middleware to access the new and old routes.

The log points to `setupLinkCheckerClient` in the client plugin (https://github.com/harlan-zw/nuxt-link-checker/blob/main/src/runtime/nuxt/plugin/view/client.ts#L40):

```
client.mjs:54 [nuxt] Calling `useRoute` within middleware may lead to misleading results. Instead, use the (to, from) arguments passed to the middleware to access the new and old routes.
useRoute @ router.js:12
setupLinkCheckerClient @ client.js:33
(anonymous) @ ui.client.js:17
Promise.then (async)
(anonymous) @ ui.client.js:16
```

### System / Nuxt Info

```Shell
- Operating System: Darwin
- Node Version:     v20.10.0
- Nuxt Version:     3.12.3
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      debug, devtools, experimental, future, typescript, app, alias, css, compatibilityDate, hooks, modules, routeRules, runtimeConfig, cookieConsent, content, dayjs, googleFonts, gtm, image, i18n, lodash, ogImage, security, site, sitemap, supabase, nitro, veeValidate, vite
- Runtime Modules:  @nuxtjs/seo@2.0.0-rc.15, @nuxt/content@2.13.1, @nuxt/image@1.7.0, @nuxtjs/google-fonts@3.2.0, @nuxtjs/i18n@8.3.1, @nuxtjs/supabase@1.3.5, @pinia/nuxt@0.5.1, @vee-validate/nuxt@4.13.2, @vueuse/nuxt@10.11.0, @weareheavy/nuxt-cookie-consent@1.0.4, @zadigetvoltaire/nuxt-gtm@0.0.13, dayjs-nuxt@2.1.9, nuxt-icon@0.6.10, nuxt-lodash@2.5.3, nuxt-security@2.0.0-rc.9, nuxt-svgo@4.0.2
- Build Modules:    -
```

I’m using `nuxtjs/seo`, and while disabling the link checker module helped, I’d love to keep using it 🙂

**Comment by andrzejkupczyk** [comment]
Thank you!

**Comment by harlan-zw** [comment]
No problem :) I have just pushed up version 3.1.0 with this fix.
